core/domain/suggestion_services.py

```diff
@@ -593,12 +619,199 @@ def get_translation_suggestions_waiting_longest_for_review_per_lang(
+                        user_contribution_rights.id):
+                    heapq.heappush(suggestions_waiting_longest_heap, (
+                        question_suggestion.last_updated, question_suggestion))
+
+        if user_contribution_rights.can_review_translation_for_language_codes:
+            for language_code in (
+                    user_contribution_rights
+                    .can_review_translation_for_language_codes):
+                # Get a list of the translation suggestions in the language code
+                # from the datastore if we haven't already gotten them.
+                if language_code not in (
+                        translation_suggestions_by_lang_code_dict):
+                    translation_suggestions_by_lang_code_dict[language_code] = (
+                        get_translation_suggestions_waiting_longest_for_review(
+                            language_code
+                        )
+                    )
+
+                translation_suggestions = (
+                    translation_suggestions_by_lang_code_dict[language_code]
+                )
+                for translation_suggestion in translation_suggestions:
+                    if len(suggestions_waiting_longest_heap) == (
+                            MAX_NUMBER_OF_SUGGESTIONS_PER_REVIEWER):
+                        # The shortest review wait time corresponds to the most
+                        # recent review submission date, which is the max of
+                        # the heap.
+                        most_recent_review_submission = max(
+                            suggestions_waiting_longest_heap)[0]
+                        # If the review submission date for the translation
+                        # suggestion is more recent than the most recent
+                        # submission date so far, we can exit early.
+                        if translation_suggestion.last_updated > (
+                                most_recent_review_submission):
+                            break
+                    # Reviewers can never review their own suggestions.
+                    elif translation_suggestion.author_id != (
+                            user_contribution_rights.id):
+                        heapq.heappush(suggestions_waiting_longest_heap, (
+                            translation_suggestion.last_updated,
+                            translation_suggestion))
+
+        # Get the key information from each suggestion that will be used to
+        # email reviewers.
+        reviewer_reviewable_suggestion_infos = []
+        for _ in python_utils.RANGE(MAX_NUMBER_OF_SUGGESTIONS_PER_REVIEWER):
+            if len(suggestions_waiting_longest_heap) == 0:
+                break
+            _, suggestion = heapq.heappop(suggestions_waiting_longest_heap)
+            reviewer_reviewable_suggestion_infos.append(
+                create_reviewable_suggestion_email_info_from_suggestion(
```

Comment:
Do your tests for this function verify that the correct snippet text is included?